### ISSUE TITLE
[BUG] mlflow.evaluate: precision_recall_auc and ROC/PR/calibration curves are wrong for a non-default pos_label

### ISSUE THREAD

**Issue by immu4989** [mlflow/mlflow#23991]
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Issues Policy acknowledgement
- [x] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?
Local machine

### MLflow version
- Client: 3.12 (current master)

### Describe the problem

When `pos_label` is set to the class that **isn't** the higher-sorted label, `mlflow.evaluate(model_type="classifier")` reports an incorrect `precision_recall_auc` and produces ROC / precision-recall / calibration **curve artifacts for the wrong class**.

The scalar `roc_auc` looks correct because ROC AUC is symmetric under a positive/negative class swap, which masks the problem — but `precision_recall_auc` is not symmetric and comes out wrong, and the logged curve plots are computed against the wrong probability column.

This is the common imbalanced-binary case where the minority/positive class is the lower-sorted label (fraud, churn, default-risk).

### Code to reproduce issue

```python
import numpy as np, pandas as pd, mlflow, sklearn.linear_model
from sklearn.metrics import average_precision_score

mlflow.set_tracking_uri("sqlite:///mlruns.db")
rng = np.random.RandomState(0)
X = rng.randn(400, 4)
y = (X.sum(1) > 0).astype(int)            # well-separated binary problem, classes {0, 1}
model = sklearn.linear_model.LogisticRegression().fit(X, y)
P = model.predict_proba(X)                # columns follow model.classes_ = [0, 1]

# Correct average-precision for positive class 0, using P(class 0):
print("correct pr_auc, pos_label=0:", round(average_precision_score(y == 0, P[:, 0]), 4))  # 0.9998

data = pd.DataFrame(X, columns=[f"f{i}" for i in range(4)]); data["label"] = y
with mlflow.start_run():
    mi = mlflow.sklearn.log_model(model, name="m")

for pl in (1, 0):
    with mlflow.start_run():
        res = mlflow.evaluate(mi.model_uri, data, targets="label", model_type="classifier",
                              evaluator_config={"pos_label": pl, "label_list": [0, 1]})
        print(f"mlflow pos_label={pl}: roc_auc={res.metrics['roc_auc']:.4f} "
              f"pr_auc={res.metrics['precision_recall_auc']:.4f}")
```

### Observed

```
correct pr_auc, pos_label=0: 0.9998
mlflow pos_label=1: roc_auc=0.9997 pr_auc=0.9997
mlflow pos_label=0: roc_auc=0.9997 pr_auc=0.3237   <-- wrong (should be ~0.9998)
```

`pos_label=1` (the higher-sorted, default-style positive class) is correct; only the non-default positive class is affected.

### What component(s) does this bug affect?
- [x] `area/evaluation`: MLflow Evaluation APIs, `mlflow.evaluate`

### Additional context

I've traced the root cause and have a fix with regression tests ready locally. Happy to open a PR — could a maintainer triage and apply the `ready` label so the PR isn't auto-closed?

**Comment by B-Step62** [comment]
Thanks for the report, please go ahead and submit a PR!
